cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -220,11 +227,12 @@ runQueryProtocolParameters (AnyConsensusModeParams cModeParams) network mOutFile
         eInMode <- toEraInMode era cMode
           & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
-        ppResult <- lift . queryExpr $ QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters
-
-        except ppResult & firstExceptT ShelleyQueryCmdEraMismatch
+        lift (queryExpr $ QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters)
+          & onLeft (throwE . ShelleyQueryCmdUnsupportedNtcVersion)
+          & onLeft (throwE . ShelleyQueryCmdEraMismatch)
```

Comment:
I was not successful in implementing the above with `hoistEither` in a way that looks pleasant.

Reply:
This is resolved.